src/sonic-yang-models/yang-models/sonic-types.yang

```diff
@@ -114,4 +118,53 @@ module sonic-types {
             pattern "percentage|used|free|PERCENTAGE|USED|FREE";
         }
     }
+
+    typedef oper-status {
+        type enumeration {
+            enum unknown;
+            enum up;
+            enum down;
+        }
+        description "Operational status of an entity such as Port, MCLAG etc";
+    }
+
+    typedef mode-status {
+        type enumeration {
+            enum enable;
+            enum disable;
+        }
+        description
+            "This type can be used where toggle functionality required.
+            For ex. IPv6-link-local-only, Dhcp-replay-link-select, SNMP traps etc";
+    }
+
+    typedef dhcp-relay-policy-action {
+        type enumeration {
+            enum discard;
+            enum append;
+            enum replace;
+        }
+        description "DHCP relay policy action value";
+    }
+
+    typedef percentage {
+        type uint8 {
+            range "0..100";
+        }
+        description
+            "Integer indicating a percentage value";
+    }
+
+    // Defined as container because it will be used as xpath in MUST/WHEN
+    // expressions in yang.
+    container operation {
```

Comment:
This will not be provided by user for any configuration. This is used for checking operation type in MUST expressions. CVL uses it while evaluating MUST expression.

Reply:
Why it is a container can you explain it better

Reply:
Container is required to use in XPath for checking operation-type in MUST expression. For ex.
        container COPP_GROUP {
            list COPP_GROUP_LIST {
                key name;
                leaf name {
                    type string;
                }
                ...
                leaf cir {
                    type uint64;
                }
                leaf cbs {
                    type uint64;
                    must "(/cmn:operation/cmn:operation = 'DELETE') or ((current()/../cir) and (current()/../cir > 0))" {
                        error-message "cbs can't be configured without cir.";
                    }
                    must "(/cmn:operation/cmn:operation = 'DELETE') or (current() >= current()/../cir)" {
                        error-message "cbs must be greater than or equal to cir";
                    }
                }

Reply:
This file is for types, I think we need to find a better place for another container. I still did not get why this is defined as container and not as a type.

Reply:
This container is also defining types of operations, so in my opinions, sonic-types.yang is best yang to place this container. Please suggest if you know other yang to place this. If this is defined only as type, then how it can be used as Xpath in MUST expression. There should be an instance whose value can be "DELETE" or "UPDATE" to validate and that instance can be created from yang leaf node.

Reply:
AFAIK, Operation team does not agree to send the operation to the switch. Also, other yang models do not use operation container. 
Moreover, from the meeting, it was not clear, why this is needed for sonic yang. Is this needed due to REST use or due to RedisDB?
Prefer to remove it right now.

Reply:
@praveen-li Again we clearly explained this is needed for the semantic validation with examples. This is needed for REST as well as gNMI. REST has the POST, PUT, PATCH and DELETE as RPCs and gNMI has the SET RPC which take create, update, delete as one of the field for any configuration request. Not sure how the controller will be able to send requests without using any of this RPCs. CVL semantic validations will not go through without this, so it is not possible to remove this.

Reply:
As asked in meeting, if this requirement is due to REST, but other yang models open-config, frr does not have operation field?.

Reply:
Yang models in open-config and frr does not have any MUST expressions for semantic validations.